cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -185,8 +186,10 @@ public void schedule(Id.Program programId, SchedulableProgramType programType, I
         throw Throwables.propagate(e);
       }
       for (Schedule schedule : schedules) {
-        String scheduleName = schedule.getName();
-        String cronEntry = schedule.getCronEntry();
+        Preconditions.checkArgument(schedule instanceof TimeSchedule);
```

Comment:
That's right, this ``WrappedScheduler`` is backed by a quartz scheduler and is meant to support only ``TimeSchedule``. In a coming PR, the ``AbstractSchedulerService`` will distribute the ``TimeSchedule`` objects to this ``WrappedScheduler``, and the ``DataSchedule`` objects to a new ``DataScheduler``.